### PR TITLE
Fix broken 'Blocked Highlight' feature in UI

### LINKED ISSUE #12182: "blocked" UI feature is broken in 2.0.0a2 and the latest releases of 1.10.x
When a DAG's number of active DagRuns >= its number of max allowed active DagRuns, it's `Schedule` will be highlighted with red colour (and a tooltip should always be added).

However this feature is broken in 2.0.0a2 and the latest a few releases of 1.10.x series.

## Example

in this case, the `schedule`'s background colour should be changed to red, according to code https://github.com/apache/airflow/blob/63ac07d9c735b1ccc8aa5fa974a260cc944cc539/airflow/www/templates/airflow/dags.html#L357-L367

While it is not working as expected.

DAGs_-_Airflow

I tried to check the history, and seems it's caused by PR https://github.com/apache/airflow/pull/6985

### PR DESCRIPTION
**Pull request by XD-DENG**
Closes https://github.com/apache/airflow/issues/12182

This "Blocked Highlight" feature helps highlight the DAGs whose # of active DagRuns is bigger than or equal to its max allowed. But this feature is broken in 2.0.0a2 and the latest releases of 1.10.x series.

## A bit more clarification

We will need the dag_id in the `class` so that code below can work as expected:

https://github.com/apache/airflow/blob/63ac07d9c735b1ccc8aa5fa974a260cc944cc539/airflow/www/templates/airflow/dags.html#L357-L367

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/www/templates/airflow/dags.html
Comment (mik-laj) [inline airflow/www/templates/airflow/dags.html]: This can cause name conflicts. I think it's worth thinking about a different selector in this case, e.g. based on the `data-dag-id` attribute.
```html
<a class="label label-default schedule" href="{{ url_for('DagRunModelView.list') }}?_flt_3_dag_id={{ dag.dag_id }}" data-dag-id="{{ dag.dag_id }}">
```
```js
vaar dag_id = "my-dag-id"
document.querySelector('[data-dag-id="'+ dag_id+'"]')
```
See: https://issues.apache.org/jira/browse/AIRFLOW-6381

Comment (XD-DENG) [reply airflow/www/templates/airflow/dags.html]: Makes sense to me. And thanks for the snippet!

Will update shortly tonight and ping you for another review.

**Comment by XD-DENG** [comment]
Hi @mik-laj , updated as you suggested earlier. Mind taking another look?


When it works the UI looks like below. I'm not sure if the colour ("`red`") matches the theme well overall or not.
I will not change it for now, and would like to leave it to folks who are better at UX.

DAGs_-_Airflow

**Comment by potiuk** [comment]
> I will not change it for now, and would like to leave it to folks who are better at UX.

Indeed. It does not look pretty :D. I am sure @ryanahamilton would like to final-touch it ;)

**Review by potiuk — APPROVED** [review]
Looks good from engineering point of view. But some final touches are needed

**Review by mik-laj — APPROVED** [review]

**Comment by XD-DENG** [comment]
I tried to change the style a bit:
- instead of changing the background colour, change the font colour
- use colour "#e43921" (the same colour of the "Delete DAG" button),  over "red".

So eventually it looks like below

DAGs_-_Airflow

And hovering over shows the title like below

image

Given other committers agree it looks ok from engineering aspect, I would like to leave the final style/theme refinement to gurus like @ryanahamilton in a separate PR.

**Comment by mik-laj** [comment]
Sound good to me. Can you create an issue?

**Comment by potiuk** [comment]
LGTM :)

**Comment by XD-DENG** [comment]
A follow-up Issue for the final style change is created at https://github.com/apache/airflow/issues/12186

**Comment by ryanahamilton** [comment]
@XD-DENG this looks good—changing the color instead of background-color looks much nicer!

**Comment by XD-DENG** [comment]
> @XD-DENG this looks good—changing the color instead of background-color looks much nicer!

Thanks @ryanahamilton ! If you don't have plan to further change the style, please feel free to close https://github.com/apache/airflow/issues/12186

**Pull request merged**
